### DISCUSSION TITLE
Datatable lazy but with Datascroller loader functionality

### DISCUSSION THREAD

**Discussion by matthijs-ll** [primefaces/community#134] [PrimeFaces] [answered]
For performance measurement, we want to get rid of pagination in our Datatables. Besides, we are not convinced that having an X amount of pages to click through is helpful when you can sort and filter. We would like users to go forth/back a few pages. We are aware of Livescrolling but that requires a restrictive container and seems to still hammer the backend when users scroll continuously.

We like the **loader** functionality of the Datascroller a lot and would like to have that on a Datatable; load the initial X rows and **add** that same amount of rows, without losing the first set, or 'page' - if you like via a 'more' button. 

Is this something we could do with the addRow method?

**Comment by Rapster** [comment]
>We like the loader functionality of the Datascroller a lot and would like to have that on a Datatable; load the initial X rows and add that same amount of rows, without losing the first set, or 'page' - if you like via a 'more' button.

You're refering to infinite scroll, which is here: https://www.primefaces.org/showcase/ui/data/datatable/scroll.xhtml (On-Demand Data)

Regarding performance, depends what your definition is... You might want to have a look at virtual scrolling (same page)

**Reply by matthijs-ll** [reply]
Thank you for thinking along. We've looked at infinite scroll and it's not suitable usability-wise, on several grounds. We're really looking for an explicit and elegant interaction pattern like that of the Datascroller loader.

**Reply by Rapster** [reply]
>  We've looked at infinite scroll and it's not suitable usability-wise, on several grounds.

Would be great to share more details with the @primefaces/primefaces-committers , what are the "grounds" ? Maybe there is something we can do about it ?

>  We're really looking for an explicit and elegant interaction pattern like that of the Datascroller loader.

I'm confused, DataTable and DataSroller does the same thing, what am I missing?

**Comment by matthijs-ll** [comment]
> > We've looked at infinite scroll and it's not suitable usability-wise, on several grounds.
> 
> Would be great to share more details with the @primefaces/primefaces-committers , what are the "grounds" ? Maybe there is something we can do about it ?
> 

We don't want a restricting container to make the scrollbar appear. There are several tables on a page which will result in a scrollable page with scrollable tables. Other than that, when looking at the demo and documentation I can't see how we can control how fast the scrolling actions trigger the fetching of new rows.

> > We're really looking for an explicit and elegant interaction pattern like that of the Datascroller loader.
> 
> I'm confused, DataTable and DataSroller does the same thing, what am I missing?

Datascroller doesn't feature sortable columns, filtering and pagination like datatables (which the latter we'd like to get rid of). So basically we would like a datatable with all its features and the option and a 'more' button which keeps the old rows in view but add new ones on a click.

**Reply by Rapster** [reply, marked as answer]
It sounds similar to this https://github.com/primefaces/primefaces/issues/7837

**Reply by matthijs-ll** [reply]
Thanks! I had found that exact question in Google but it got me to the old forum. We're looking into his workaround.
